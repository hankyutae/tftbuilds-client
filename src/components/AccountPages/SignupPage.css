.signup-box {
  height: 450px;
  width: 300px;/* 
  background: rgba(200, 200, 200,0.88); */

  position: absolute;
  top:150px;
  border-radius:5px;
  z-index:2;
}


.offset1signin{
  stop-color:#FDE1BC;
  stop-opacity:1;
}

.offset2signin{
  stop-color:rgb(251,175,71);
  stop-opacity:1;
}


.hex-bronze{
  fill:rgba(112, 26, 26,.8);
}
.hex-silver{
  fill:#FBAF47;
}

.signup-box input{
  
  border-radius:4px;
}

@media screen and (min-width:300px){
  .signup-box{
    left: calc(50% - 150px);
  }
}


.signup-box{
  padding:12px 10px;
}

.signup-box .fieldset-style{
padding:0;
margin:0;
text-align:left;
border:0;
} 

.signup-box .fieldset-style > legend{
margin-bottom: 2%;
}

.signup-box input{
width:100%;
}
.signup-box .signup-sign-up{
width:100%;
}

.signup-box .signup-sign-up{
  width:100%;
  border:0px;
  padding:0.5rem;
  border-radius:5px;
  background-color:#A2B9B1;
  background-color:#e8e5f3;
  font-weight:550;
}


@media (hover:hover){
  .signup-box .signup-sign-up:hover{
    background-color:rgba(162,185,177,0.8);
  }
}
.signup-box input{
  padding:0.3rem;
  border:0px;
}

.signup-box h1.signup-box-logo{
padding:0;
margin:0;
/* position:absolute;
top: -50px;
left:67px; */
margin-bottom:0px;

}/* 
@media screen and (max-height:499px){
  main{
    min-height:600px;
  }
} */