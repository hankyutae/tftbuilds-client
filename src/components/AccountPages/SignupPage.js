import React from 'react';
import './SignupPage.css';
import AuthApiService from '../../services/auth-api-service';


export default class SignupPage extends React.Component{


  state = { error: null }

  handleSubmit = ev => {
    ev.preventDefault()
    const { email, user_name, password, passwordRepeat} = ev.target;
    if(passwordRepeat.value!==password.value){
      return this.setState({
        error:'Passwords must match'
      })
    }

    this.setState({ error: null })
    AuthApiService.postUser({
      user_name: user_name.value,
      password: password.value,
      email: email.value,
    })
      .then(user => {
        email.value = ''
        user_name.value = ''
        password.value = ''
        this.onRegistrationSuccess()
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }
  onRegistrationSuccess = user => {
    const { history } = this.props
    history.push('/login')
  }
  render(){
    const {error}=this.state;
    return(
      <form className='signup-box' onSubmit={this.handleSubmit}>
      <h1 className="signup-box-logo">
        TFT Comps
      </h1>
      <fieldset className='fieldset-style'>
          <legend>Sign up </legend>
          <div role='alert'>
          {error && <p className='red'>{error}</p>}
          </div>
          <p>
            <label htmlFor="signup-userid">Username</label><br/>
            <input type="text" name="user_name" id="signup-userid"/>
          </p>
          <p>
            <label htmlFor="signup-email">Email</label><br/>
            <input type="email" name="email" id="signup-email"/>
          </p>
          <p>
            <label htmlFor="signup-password">Password</label><br/>
            <input id="signup-password" name="password" type="password"/>
          </p>
          <p>
            <label htmlFor="signup-password-repeat">Repeat Password</label><br/>
            <input id="signup-password-repeat" name="passwordRepeat" type="password"/>
          </p>
      </fieldset>
      <button className="signup-sign-up"> Sign up </button> <br/>
      
    </form>
    );
  }
}
