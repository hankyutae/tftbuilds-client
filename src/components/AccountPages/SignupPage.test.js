import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import SignupPage from './SignupPage';
import renderer from "react-test-renderer";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <SignupPage/>
    </BrowserRouter>
  , div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create( 
      <BrowserRouter>
        <SignupPage/>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});