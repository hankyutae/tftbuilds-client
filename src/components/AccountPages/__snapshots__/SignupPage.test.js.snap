// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <div
    className="svg-div"
  >
    <svg
      className="svg-gold"
      height="600"
      width="310"
    >
      <defs>
        <radialGradient
          cx="50%"
          cy="50%"
          fx="50%"
          fy="50%"
          id="grad2"
          r="90%"
        >
          <stop
            className="offset1signin"
            offset="0%"
          />
          <stop
            className="offset2signin"
            offset="100%"
          />
        </radialGradient>
      </defs>
      <polyline
        className="hex-bronze"
        points="150,0 300,87 300,97 150,10 0,97 0,87 150,0"
      />
       /&gt;
      <polyline
        className="hex-bronze"
        points="150,556 300,470 300,480 150,566 0,480 0,470 150,556"
      />
      <polygon
        fill="url(#grad2)"
        points="150,10 300,97 300,470 150,556 0,470 0,97 150,10"
      />
    </svg>
  </div>,
  <form
    className="signup-box"
    onSubmit={[Function]}
  >
    <h1
      className="signup-box-logo"
    >
      Register
    </h1>
    <fieldset
      className="fieldset-style"
    >
      <div
        role="alert"
      />
      <p>
        <label
          htmlFor="signup-userid"
        >
          Username
        </label>
        <br />
        <input
          id="signup-userid"
          name="user_name"
          type="text"
        />
      </p>
      <p>
        <label
          htmlFor="signup-email"
        >
          Email
        </label>
        <br />
        <input
          id="signup-email"
          name="email"
          type="email"
        />
      </p>
      <p>
        <label
          htmlFor="signup-password"
        >
          Password
        </label>
        <br />
        <input
          id="signup-password"
          name="password"
          type="password"
        />
      </p>
      <p>
        <label
          htmlFor="signup-password-repeat"
        >
          Repeat Password
        </label>
        <br />
        <input
          id="signup-password-repeat"
          name="passwordRepeat"
          type="password"
        />
      </p>
    </fieldset>
    <button
      className="signup-sign-up"
    >
       Sign up 
    </button>
     
    <br />
  </form>,
]
`;
